service_container.rst

```diff
@@ -607,6 +607,23 @@ accessor methods for parameters::
     To learn more about compiling the container see
     :doc:`/components/dependency_injection/compilation`.
 
+It is also possible to deprecate a container parameter. This is particularly useful
+if you're writing your own package and want to warn developers to not use a parameter anymore
+because it will be removed in one of the next version of your package.
+Deprecating a container parameter can be done as follow::
+
+    $container->deprecateParameter('vendor.foo', 'vendor-name/package-name', '1.3', '"vendor.foo" is deprecated, you should use "vendor.foo_replacement" instead.');
```

Comment:
Maybe replacing `vendor.foo` to `vendor.old_name` is more explicit ?

I think we can remove note by adding optionally comment (it's done in many place in doc) WDYT ?

```suggestion
    $container->deprecateParameter(
        'vendor.old_param',
        'vendor-name/package-name',
        '1.3',
        // optionally you can change message, by default it will be 'The parameter "vendor.old_param" is deprecated.'
        '"vendor.old_param" is deprecated, you should use "vendor.new_param" instead.'
    );
```

Reply:
About `vendor.old_name` I'm not sure, as this could be a parameter that will simply disappear without any "new name". 🤔 

However, I think that indenting the example is a good idea! I'm doing it.

Finally, about the comment above the optional message, I would say I'm 👍 on it, but it seems a bit long however. But it may be just me, I'll check examples in the doc!

Reply:
Is `'vendor-name/package-name', '1.3',` on the same line is a choice ?

Reply:
Nope it's not, updated, thanks! 😄